### PR TITLE
Refactor common scanner logic into separate scanner-global-library package

### PR DESCRIPTION
**Pull request by lamaks**
#### Description of changes

Refactor common scanner logic into separate scanner-global-library package
Integrate scanner-global-library package in crawler package
Refactor logger container setup

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: packages/scanner-global-library/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<!--
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+-->
+
+The `scanner-global-library` package contains the common web page scanning logic used by both service and command line packages.
```
Comment (AhmedAbdoOrtiga) [inline packages/scanner-global-library/README.md:6]: I don't see how it's used in the service package

File: packages/scanner-global-library/src/page-handler.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { inject, injectable } from 'inversify';
+import { GlobalLogger } from 'logger';
+import { Page } from 'puppeteer';
+
+@injectable()
+export class PageHandler {
```
Comment (AhmedAbdoOrtiga) [inline packages/scanner-global-library/src/page-handler.ts:8]: is this class used anywhere?

File: packages/crawler/src/page-processors/page-processor-base.ts [hunk 3]
```diff
@@ -68,15 +71,35 @@ export abstract class PageProcessorBase implements PageProcessor {
...
+
+            let response: Response;
+            try {
+                response = await this.gotoExtended(inputs.page, inputs.request, {
+                    waitUntil: 'networkidle0',
+                    timeout: this.gotoTimeoutSecs * 1000,
+                });
+                // Catch only URL navigation error here
+            } catch (err) {
+                const navigationError = this.pageResponseProcessor.getNavigationError((err as Error).message);
```
Comment (AhmedAbdoOrtiga) [inline packages/crawler/src/page-processors/page-processor-base.ts:85]: Let's keep the error trace in the browserError

File: packages/crawler/src/page-processors/page-processor-base.ts [hunk 4]
```diff
@@ -125,7 +148,11 @@ export abstract class PageProcessorBase implements PageProcessor {
         await this.blobStore.setValue(`${scanData.id}.data`, scanData);
     }
 
-    protected async logPageError(request: Apify.Request, error: Error): Promise<void> {
+    protected async logBrowserFailure(request: Apify.Request, browserError: BrowserError): Promise<void> {
```
Comment (AhmedAbdoOrtiga) [inline packages/crawler/src/page-processors/page-processor-base.ts:151]: we need to have the same function, only logError, the file should be the error trace (more details), error type column would be error type, and errorDescription column will be error.message

File: packages/scanner-global-library/src/browser-error.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    | 'ResourceLoadFailure'
+    | 'InvalidUrl'
+    | 'EmptyPage'
+    | 'HttpErrorCode'
+    | 'NavigationError'
+    | 'InvalidContentType'
+    | 'UrlNotResolved'
+    | 'ScanTimeout';
+
+export interface BrowserError {
```
Comment (AhmedAbdoOrtiga) [inline packages/scanner-global-library/src/browser-error.ts:16]: can you add errorTrace?

**Review by AhmedAbdoOrtiga — CHANGES_REQUESTED** [review]
- Can you add "feat(cli-summary-report):" in the PR title.
- Do we need to incorporate "PageResponseProcessor" in file scanning?

File: packages/scanner-global-library/README.md (hunk 1 above)
Comment (lamaks) [reply packages/scanner-global-library/README.md:6]: Right, I'll add this package to the service in next PR

File: packages/crawler/src/page-processors/page-processor-base.ts (hunk 3 above)
Comment (lamaks) [reply packages/crawler/src/page-processors/page-processor-base.ts:85]: ok

File: packages/scanner-global-library/src/browser-error.ts (hunk 5 above)
Comment (lamaks) [reply packages/scanner-global-library/src/browser-error.ts:16]: ok

File: packages/crawler/src/page-processors/page-processor-base.ts (hunk 4 above)
Comment (lamaks) [reply packages/crawler/src/page-processors/page-processor-base.ts:151]: added error stack to browser error

File: packages/scanner-global-library/src/page-handler.ts (hunk 2 above)
Comment (lamaks) [reply packages/scanner-global-library/src/page-handler.ts:8]: Correct, I'll add in next PR to unblock this one

**Review by AhmedAbdoOrtiga — APPROVED** [review]

**Pull request merged**